File: app/src/main/java/app/axon/test/ui/base/BaseActivity.java
package app.axon.test.ui.base;

import android.app.ProgressDialog;
import android.widget.Toast;

import java.lang.ref.WeakReference;

import androidx.appcompat.app.AppCompatActivity;
import app.axon.test.R;
import app.axon.test.api.RetrofitClient;

public abstract class BaseActivity extends AppCompatActivity {

    private WeakReference<ProgressDialog> progressDialog;

    public RetrofitClient getRetrofitClient() {
        return new RetrofitClient(this);
    }

    public void showProgress() {
        if (progressDialog == null) {
            progressDialog = new WeakReference<>(new ProgressDialog(this));
            progressDialog.get().setTitle(getString(R.string.progress_title));
            progressDialog.get().setMessage(getString(R.string.progress_message));
            progressDialog.get().setCanceledOnTouchOutside(false);
            progressDialog.get().setCancelable(false);
        }

        progressDialog.get().show();
    }

    public void hideProgress() {
        if (this.isDestroyed() || this.isFinishing()) return;
        if (progressDialog != null && progressDialog.get() != null && progressDialog.get().isShowing()) {
            progressDialog.get().dismiss();
            progressDialog = null;
        }
    }

    public void showMessage(String message) {
        hideProgress();
        if (!this.isFinishing()) {
            Toast.makeText(getBaseContext(), message, Toast.LENGTH_SHORT).show();
        }
    }

    public void showMessage(int message) {
        hideProgress();
        if (!this.isFinishing()) {
            Toast.makeText(getBaseContext(), message, Toast.LENGTH_SHORT).show();
        }
    }

}

File: app/src/main/java/app/axon/test/ui/base/BaseActivity.kt
package app.axon.test.ui.base

import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import app.axon.test.api.RetrofitClient


abstract class BaseActivity : AppCompatActivity(), StandartDialogs {


    fun getRetrofitClient(): RetrofitClient {
        return RetrofitClient(baseContext)
    }

    override fun showToast(message: String) {
        if (!this.isFinishing) {
            Toast.makeText(baseContext, message, Toast.LENGTH_SHORT).show()
        }
    }

    override fun showToast(message: String, length: Int) {
        if (!this.isFinishing) {
            Toast.makeText(baseContext, message, length).show()
        }
    }

    override fun showMessage(message: String) {
        showToast(message, Toast.LENGTH_LONG)
    }

    override fun showMessage(message: Int) {
        showToast(getString(message), Toast.LENGTH_LONG)
    }
}
